Src/Support/Google.Apis.Auth.AspNetCore/GoogleOpenIdConnectExtensions.cs

```diff
@@ -0,0 +1,135 @@
+*/
+
+using Google.Apis.Auth.AspNetCore;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using System;
+using System.Linq;
+
+// Microsoft recommend that all service Add methods go in this namespace.
+// See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class GoogleOpenIdConnectExtensions
+    {
+        public static AuthenticationBuilder AddGoogleOpenIdConnect(this AuthenticationBuilder builder) =>
+                        AddGoogleOpenIdConnect(builder, _ => { });
+
+        public static AuthenticationBuilder AddGoogleOpenIdConnect(this AuthenticationBuilder builder,
+           Action<OpenIdConnectOptions> configureOptions) =>
+               AddGoogleOpenIdConnect(builder, GoogleOpenIdConnectDefaults.AuthenticationScheme, configureOptions);
+
+        public static AuthenticationBuilder AddGoogleOpenIdConnect(this AuthenticationBuilder builder,
+           string authenticationScheme, Action<OpenIdConnectOptions> configureOptions) =>
+               AddGoogleOpenIdConnect(builder, authenticationScheme, GoogleOpenIdConnectDefaults.DisplayName, configureOptions);
+
+        public static AuthenticationBuilder AddGoogleOpenIdConnect(this AuthenticationBuilder builder,
+            string authenticationScheme, string displayName, Action<OpenIdConnectOptions> configureOptions)
+        {
+            // Service to provide the authentication scheme name.
+            builder.Services.AddSingleton(new GoogleAuthenticationSchemeProvider(authenticationScheme));
+            // Services to facilitate the GoogleScopedAuthorize attribute.
+            builder.Services.AddTransient<IAuthorizationPolicyProvider, GoogleScopedPolicyProvider>();
+            builder.Services.AddScoped<IAuthorizationHandler, GoogleScopedAuthorizeHandler>();
+            // Required to provide access to HttpContext in GoogleAuthProvider.
+            builder.Services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            // Service to provide user access to the Google auth information.
+            builder.Services.AddSingleton<IGoogleAuthProvider, GoogleAuthProvider>();
+            return builder.AddOpenIdConnect(authenticationScheme, displayName, options =>
+            {
+                // Google's OpenID authority URL.
+                options.Authority = "https://accounts.google.com";
+                // Response-code to get an id-token with profile information,
+                // and a code to use to get an access-token and refresh-token.
+                options.ResponseType = "id_token code";
+                // Scopes to announce this is an OpenID auth, and get simple profile information.
```

Comment:
Will there only ever be one OpenIdConnect provider? If multiple providers clear options.Scope, that sounds bad. Admittedly I don't understand all of this in detail.

Reply:
There will only ever be one that uses this instance of `options`.